channels/channel_state.go

```diff
@@ -110,32 +116,54 @@ func (c channelState) Message() string {
 
 func (c channelState) Vouchers() []datatransfer.Voucher {
 	vouchers := make([]datatransfer.Voucher, 0, len(c.ic.Vouchers))
-	for _, encoded := range c.ic.Vouchers {
+	for i, encoded := range c.ic.Vouchers {
+		if c.decodedVouchers[i] != nil {
+			vouchers = append(vouchers, *c.decodedVouchers[i])
+			continue
+		}
 		decoder, _ := c.voucherDecoder(encoded.Type)
 		encodable, _ := decoder.DecodeFromCbor(encoded.Voucher.Raw)
-		vouchers = append(vouchers, encodable.(datatransfer.Voucher))
+		voucher := encodable.(datatransfer.Voucher)
+		c.decodedVouchers[0] = &voucher
```

Comment:
```suggestion
		c.decodedVouchers[i] = &voucher
```

looks like a copypasta bug?